public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx

```diff
@@ -83,43 +84,89 @@ export const PromQueryBuilder = React.memo<Props>((props) => {
       .map((k) => ({ value: k }));
   };
 
+  const getLabelValuesAutocompleteSuggestions = (
+    queryString?: string,
+    labelName?: string
+  ): Promise<SelectableValue[]> => {
+    const forLabel = {
+      label: labelName ?? '__name__',
+      op: '=~',
+      value: regexifyLabelValuesQueryString(`${queryString}`),
+    };
+    const labelsToConsider = query.labels.filter((x) => x.label !== forLabel.label);
+    labelsToConsider.push(forLabel);
+    if (query.metric) {
+      labelsToConsider.push({ label: '__name__', op: '=', value: query.metric });
+    }
+    const expr = promQueryModeller.renderLabels(labelsToConsider);
+
+    let response;
+    if (datasource.hasLabelsMatchAPISupport()) {
+      response = getLabelValuesFromLabelValuesAPI(forLabel, expr);
+    } else {
+      response = getLabelValuesFromSeriesAPI(forLabel, expr);
+    }
+
+    return response.then((response: SelectableValue[]) => {
+      if (response.length > PROMETHEUS_QUERY_BUILDER_MAX_RESULTS) {
```

Comment:
We're doing this little splice all over the place, do we want a helper function to clean these up?

Reply:
Great idea Galen, but maybe we tackle that and the types in a follow up PR?

Reply:
:rofl:

Reply:
Added in #58375: https://github.com/grafana/grafana/issues/58375#issuecomment-1306158748